src/Widgets/TerminalWidget.vala

```diff
@@ -42,12 +42,13 @@ namespace Terminal {
             }
         }
 
-        private Gtk.Menu menu {
+        private Menu context_menu_model {
             get {
-                return main_window.menu;
+                return main_window.context_menu_model;
             }
         }
 
+        private Gtk.PopoverMenu context_popover;
```

Comment:
This should probably not be `Gtk.PopoverMenu` in Gtk3, but instead `Gtk.Menu` and we can change it to `PopoverMenu` in Gtk4